jobs/payment-jobs/tasks/cfs_create_invoice_task.py

```diff
@@ -49,6 +53,92 @@ def create_invoices(cls):
         cls._create_online_banking_invoices()
         cls._create_eft_invoices()
         cls._create_wire_invoices()
+        cls._create_rs_invoices()
+
+    @classmethod
+    def _create_rs_invoices(cls):  # pylint: disable=too-many-locals
+        """Create RS invoices in to CFS system."""
+        # Find all pending routing slips.
+
+        # find all routing slip invoices [cash or cheque]
+        # create invoices in csf
+        # do the recipt apply
+        invoices: List[InvoiceModel] = InvoiceModel.query \
+            .filter(InvoiceModel.payment_method_code == PaymentMethod.INTERNAL.value) \
+            .filter(InvoiceModel.invoice_status_code == InvoiceStatus.APPROVED.value) \
+            .filter(InvoiceModel.routing_slip is not None) \
+            .order_by(InvoiceModel.created_on.asc()).all()
+
+        current_app.logger.info(f'Found {len(invoices)} to be created in CFS.')
+        receipt_response = {}
+        for invoice in invoices:
+            # Create a CFS invoice
+            current_app.logger.debug(f'Creating cfs invoice for invoice {invoice.id}')
+            routing_slip = RoutingSlipModel.find_by_number(invoice.routing_slip)
+            routing_slip_payment_account: PaymentAccountModel = PaymentAccountModel.find_by_id(
+                routing_slip.payment_account_id)
+
+            # apply invoice to the active CFS_ACCOUNT which will be the parent routing slip
+            active_cfs_account = CfsAccountModel.find_effective_by_account_id(
+                routing_slip_payment_account.id)
+
+            invoice_response = CFSService.create_account_invoice(transaction_number=invoice.id,
+                                                                 line_items=invoice.payment_line_items,
+                                                                 cfs_account=active_cfs_account)
+            invoice_number = invoice_response.json().get('invoice_number', None)
+            receipt_applied_amount = 0
+            # a routing slip can have multiple receipts
+            cfs_accounts: List[CfsAccountModel] = CfsAccountModel.find_by_account_id(
+                routing_slip_payment_account.id)
+            # an invoice has to be applied to multiple receipts ; apply till the balance is zero
+            for cfs_account in cfs_accounts:
```

Comment:
And this I think should be reverse. While linking an invoice; update the payment account reference for the child invoice. Then here; get all the routing slips (which is considered as receipt in CFS).